repository/users/user.go

```diff
@@ -0,0 +1,17 @@
+package users
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/ivanterekh/go-skeleton/model"
+)
+
+// ErrNoSuchUser is the error returned by
+// functions that search for users.
+var ErrNoSuchUser = errors.Errorf("no such user")
+
+// Repository is an interface for user searching.
+type Repository interface {
+	ByCreds(email, password string) (*model.User, error)
```

Comment:
Add comments to interface methods

Also name of methods not ideal imho. Get<> or Search<> - depends on what they are doing